src/deferred.rs

```diff
@@ -0,0 +1,166 @@
+use std::mem;
+use std::ptr;
+
+/// Provides methods to dispatch a call to a `FnOnce()` from a trait object.
```

Comment:
Why do we need `Callback` trait`?  Can we inline the implementation for `FnOnce` in the definition of `Deferred`?

Reply:
Indeed - it's not needed. I've simplified the implementation by removing the trait.